### PR TITLE
Post about being a reverse dependencies

### PR DESCRIPTION
**Pull request by thomasp85**
This is a draft, please comment on the content and wording

### CODE REVIEW

**Review by batpigandme — APPROVED** [review]
This is really good, Thomas. Most of my edits are commas, plus a grammatical change or two. 🙂

File: content/blog/playing-on-the-same-team-as-your-dependecy/index.Rmd [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+can access as a user. There is practically no data or function you can't
+access and modify which makes the concept of APIs a question of
+conventions. Those conventions are quite well defined when it comes to
+functions in packages, but much less so for everything else. We will
+discuss functions first and then proceed into the more gray area of
+objects and data.
+
+### Exported functions
+
+When creating a package you are now required to provide a NAMESPACE file
```
[the remark this answers is not in the document]
Comment (thomasp85) [reply content/blog/playing-on-the-same-team-as-your-dependecy/index.Rmd:83]: Not that new... def in the past :-)

**Review by jennybc — COMMENTED** [review]
One thing I wrote recently that I'd love to repeat here if it fits: flaky tests suck for _everyone_.

https://r-pkgs.org/testing-advanced.html#flaky-tests

Really this point:

> Finally, flaky tests cause problems for the maintainers of your dependencies. When the packages you depend on are updated, CRAN runs `R CMD check` on all reverse dependencies, including your package. If your package has flaky tests, your package can be the reason another package does not clear CRAN’s incoming checks and can delay its release.

Caveat: I think this issue of test flakiness is separate from the main themes of this post. But, as I know you are painfully aware, pure flakiness is really costly when you have several hundred or several thousand revdeps. With enough revdeps, you're almost guaranteed to get some false positives (check failures due to flakiness and not, say, the ggplot2 release). I'm not sure if package authors with fewer revdeps appreciate this and that other maintainers then need to analyze those test failures and make the case to CRAN that they are false positives.

**Comment by thomasp85** [comment]
@jennybc I've added a final section that kind of underlines your point (though using slightly different wording to fit with the post) — how do you feel about that?

**Pull request merged**